cpp/src/arrow/util/align_util.h

```diff
@@ -64,5 +66,23 @@ inline BitmapWordAlignParams BitmapWordAlign(const uint8_t* data, int64_t bit_of
   return p;
 }
 
+template <typename T>
```

Comment:
Why is this templated?  Right now this relies on `T::data()` existing.  What other types of `T` do you expect to accept?